File: frontend/src/services/address.ts
import api from './api';

export interface AddressRequest {
  receiverName: string;
  phone: string;
  line1: string;
  ward?: string;
  district?: string;
  city: string;
  lat?: number;
  lng?: number;
  isDefault?: boolean;
}

export interface AddressDTO {
  id: number;
  receiverName: string;
  phone: string;
  line1: string;
  ward?: string;
  district?: string;
  city: string;
  lat?: number;
  lng?: number;
  isDefault: boolean;
}

export const listMyAddresses = async (userId: number): Promise<AddressDTO[]> => {
  const res = await api.get(`/addresses`, { params: { userId } });
  return res.data;
};

export const getDefaultAddress = async (userId: number): Promise<AddressDTO | null> => {
  try {
    const res = await api.get(`/addresses/default`, { params: { userId } });
    return res.data;
  } catch (e: any) {
    if (e.response?.status === 404) return null;
    throw e;
  }
};

export const createAddress = async (userId: number, payload: AddressRequest): Promise<AddressDTO> => {
  const res = await api.post(`/addresses`, payload, { params: { userId } });
  return res.data;
};

export const updateAddress = async (userId: number, id: number, payload: AddressRequest): Promise<AddressDTO> => {
  const res = await api.put(`/addresses/${id}`, payload, { params: { userId } });
  return res.data;
};

export const setDefaultAddress = async (userId: number, id: number): Promise<AddressDTO> => {
  const res = await api.put(`/addresses/${id}/default`, null, { params: { userId } });
  return res.data;
};

File: backend/src/main/java/com/fastfood/management/entity/Address.java
package com.fastfood.management.entity;

import jakarta.persistence.*;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "addresses")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Address {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;
    
    @Column(name = "receiver_name", nullable = false)
    private String receiverName;
    
    @Column(nullable = false)
    private String phone;
    
    @Column(nullable = false)
    private String line1;
    
    private String ward;
    
    private String district;
    
    @Column(nullable = false)
    private String city;
    
    private Double lat;
    
    private Double lng;
    
    // Convenience methods for drone delivery
    public Double getLatitude() {
        return lat;
    }
    
    public Double getLongitude() {
        return lng;
    }
    
    @Column(name = "is_default")
    @Builder.Default
    private boolean isDefault = false;
}